illumos-utils/src/svcs.rs

```diff
@@ -0,0 +1,333 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Utilities for reporting SMF services' status.
+
+use crate::ExecutionError;
+#[cfg(target_os = "illumos")]
+use crate::PFEXEC;
+#[cfg(target_os = "illumos")]
+use crate::execute_async;
+#[cfg(target_os = "illumos")]
+use crate::zone::SVCS;
+
+use chrono::DateTime;
+use chrono::Utc;
+use schemars::JsonSchema;
+use serde::Deserialize;
+use serde::Serialize;
+use slog::Logger;
+use slog::{error, info};
+use std::fmt::Display;
+#[cfg(target_os = "illumos")]
+use tokio::process::Command;
+
+/// Wraps commands for interacting with interfaces.
+pub struct Svcs {}
+
+impl Svcs {
+    /// Lists SMF services that are enabled but not running
+    #[cfg(target_os = "illumos")]
+    pub async fn in_maintenance(
+        log: &Logger,
+    ) -> Result<SvcsInMaintenanceResult, ExecutionError> {
+        let mut cmd = Command::new(PFEXEC);
+        let cmd = cmd.args(&[SVCS, "-Za", "-H", "-o", "state,fmri,zone"]);
+        let output = execute_async(cmd).await?;
```

Comment:
Tiny nit - I think we've occasionally seems some weird issues with hung processes, so it might be helpful to drop `info!`s before / after this execution to get a breadcrumb to help narrow that down.